l1-contracts/contracts/zksync/facets/Executor.sol

```diff
@@ -477,4 +486,44 @@ contract ExecutorFacet is Base, IExecutor {
     function _setBit(uint256 _bitMap, uint8 _index) internal pure returns (uint256) {
         return _bitMap | (1 << _index);
     }
+
+    /// @notice Calls the point evaluation precompile and verifies the output
+    // Verify p(z) = y given commitment that corresponds to the polynomial p(x) and a KZG proof.
+    // Also verify that the provided commitment matches the provided versioned_hash.
+    function _pointEvaluationPrecompile(bytes32 _versionedHash, bytes32 _inputPoint, bytes calldata _valueCommitmentProof) internal returns (bytes32) {
+        bytes memory precompileInput = abi.encodePacked(_versionedHash, _inputPoint, _valueCommitmentProof);
+
+        (bool success, bytes memory data) = POINT_EVALUATION_PRECOMPILE_ADDR.call(precompileInput);
+
+        require(success, "failed to call point evaluation precompile");
+        uint256 result = abi.decode(data, (uint256));
+        require(result == BLS_MODULUS, "precompile unexpected output");
+    }
+
+    function _verifyBlobInformation(bytes calldata _pubdataCommitments, bytes32 _blobLinearHash) internal returns (bytes32 blobCommitmentRollingHash) {
+        uint256 versionedHashIndex = 0;
+
+        for (uint256 i = 0; i < _pubdataCommitments.length; i += POINT_EVALUATION_INPUT_SIZE) {
```

Comment:
I know that it is on draft, but just in case, if `_pubdataCommitments` should be non-empty. Otherwise, the loop won't even make 1 iteration, and the point evaluation check will be bypassed.